### PR TITLE
Rename local_lcos and basic_execution modules

### PR DESCRIPTION
**Pull request by aurianer**
Fixes partially #4594

- Rename `local_lcos` to `lcos_local`
- Rename `basic_execution` to `execution_base`

### CODE REVIEW

File: docs/sphinx/releases/whats_new_1_4_0.rst [hunk 1]
```diff
@@ -261,7 +261,7 @@ Closed pull requests
 * :hpx-pr:`4179` - Moving ``zip_iterator`` and ``transform_iterator`` to
   iterator_support module
 * :hpx-pr:`4178` - Move checkpointing support to its own module
-* :hpx-pr:`4177` - Small const fix to ``basic_execution`` module
```
[the remark this answers is not in the document]
Comment (aurianer) [reply docs/sphinx/releases/whats_new_1_4_0.rst:264]: oh yes totally !

File: libs/execution_base/CMakeLists.txt [hunk 2]
```diff
@@ -6,36 +6,36 @@
...
+    hpx/execution_base/register_locks.hpp
+    hpx/execution_base/resource_base.hpp
+    hpx/execution_base/this_thread.hpp
 )
 
-set(basic_execution_compat_headers
+set(execution_base_compat_headers
     hpx/basic_execution.hpp hpx/parallel/executors/execution_fwd.hpp
     hpx/synchronization/detail/yield_k.hpp hpx/util/register_locks.hpp
     hpx/util/yield_while.hpp
```
[the remark this answers is not in the document]
Comment (aurianer) [reply libs/execution_base/CMakeLists.txt:23]: Actually I already added those but I forgot to add their names in this `CMakeLists.txt`
I used the compatibility header script so they point to their respective `execution_base` file, should I change this to make them point to `hpx/modules/execution_base.hpp` or is this fine like this ?

[the remark this answers is not in the document]
Comment (msimberg) [reply libs/execution_base/CMakeLists.txt:25]: Let's leave them like this in this PR. I might change them in my PR for #4639, not sure yet.

**Comment by msimberg** [comment]
I should have my merging rights revoked... I meant to merge https://github.com/STEllAR-GROUP/hpx/pull/4753, and merged this instead when going through my tabs. There were still some inspect problems which we'll fix on master. Apologies for this (but at least this is in now!).

**Pull request merged**